processor/processor.go

```diff
@@ -1205,6 +1191,10 @@ func (proc *Handle) processJobsForDest(subJobs subJob, parsedEventList [][]types
 			// Iterate through all the events in the batch
 			for eventIndex, singularEvent := range singularEvents {
 				messageId := misc.GetStringifiedData(singularEvent["messageId"])
+				sampleEvent, err := jsonfast.Marshal(singularEvent)
```

Comment:
Out of curiosity, how big can this `singularEvent` be?

Reply:
It would be the event which we receive in the gateway so max 4MB size

Reply:
With your change they end up in a map though for batching right? Have you checked whether this could create an issue in terms of sudden memory spikes? I'm asking because it looks like we had `{}` hard-coded before so what if now we get a bunch of 4MB events all in a row?